ext/mbstring/mbstring.c

```diff
@@ -3422,48 +3414,33 @@ static zend_string* html_numeric_entity_decode(zend_string *input, const mbfl_en
 							value = (value * 16) + 10 + (w - 'A');
 						}
 					}
-					uint32_t original;
-					if (html_numeric_entity_deconvert(value, convmap, mapsize, &original)) {
-						*converted++ = original;
+					if (html_numeric_entity_deconvert(value, convmap, mapsize, converted)) {
+						converted++;
 						if (*p2 == ';')
 							p2++;
 					} else {
 						memcpy(converted, p, (p2 - p) * 4);
 						converted += p2 - p;
 					}
 				}
-			} else if (p2 == wchar_buf + out_len && in_len) {
```

Comment:
Correct handling of this case now falls out automatically from the 'Possible decimal entity' section below.